crates/circuit/src/dag_node.rs

```diff
@@ -0,0 +1,329 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::circuit_instruction::CircuitInstruction;
+use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
+use pyo3::types::{PyDict, PyList, PySequence, PyString, PyTuple};
+use pyo3::{intern, PyObject, PyResult};
+
+static SEMANTIC_EQ_PYTHON: GILOnceCell<PyResult<PyObject>> = GILOnceCell::new();
+
+/// Parent class for DAGOpNode, DAGInNode, and DAGOutNode.
+#[pyclass(module = "qiskit._accelerate.circuit", subclass)]
+#[derive(Clone, Debug)]
+pub struct DAGNode {
+    #[pyo3(get, set)]
+    pub _node_id: isize,
+}
```

Comment:
Can we type `_node_id` as the `petgraph::NodeIndex` that it represents, so it's well-typed from within Rust space?  I imagine we can't define the PyO3 conversion traits on it because of the coherence rules, but we could manually implement the Python-space getters and setters?

Reply:
It sounds like we don't mind too much if this stays an `isize` since we'll remove it completely in the future, so I'll mark this resolved.